.github/workflows/publish.yaml

```diff
@@ -10,6 +10,11 @@ on:
       - integrations/viem-v2/v[0-9]+.[0-9]+.[0-9]+*
       - integrations/wagmi-v2/v[0-9]+.[0-9]+.[0-9]+*
 
+# See: https://docs.npmjs.com/generating-provenance-statements
+# See: https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds
+permissions:
+  id-token: write # to enable use of OIDC for npm provenance
```

Comment:
`npm-publish` does the same thing.

https://github.com/JS-DevTools/npm-publish/blob/e06fe3ef65499b38eb12224f2a60979f6d797330/.github/workflows/ci-cd.yaml#L213